lib/cocoapods/sandbox/headers_store.rb

```diff
@@ -51,7 +51,9 @@ def initialize(sandbox, relative_path, visibility_scope)
       #
       def search_paths(platform, target_name = nil, use_modular_headers = false)
         key = SEARCH_PATHS_KEY.new(platform.name, target_name, use_modular_headers)
-        return @search_paths_cache[key] if @search_paths_cache.key?(key)
```

Comment:
this introduced an extra hash lookup and this table can grow quite large on larger projects. this function is also called a high number of times